raster-foundry/raster-foundry
# Add TracingContext to BacksplashMosaic

## Overview

In the first pass of adding tracing to backsplash tracing contexts were added to `BacksplashImage` - in a few cases this wasn't ideal as the nesting for mosaics weren't 100 percent and also we couldn't add tracing to a few areas of the response cycle. This PR removes tracing context from images and adds it to mosaic. Most methods on images already took a tracing context as an argument, the difference now is that we can use child contexts from the mosaic tracing context to better group traces.

### Checklist


### Demo
![image](https://user-images.githubusercontent.com/898060/65692167-1a94d100-e040-11e9-880d-7bd10699a3c3.png)

### Notes

Still mostly focused on project layer mosaics since that's where most requests come in.

## Testing Instructions

- Get a JWT token and export it to your environment: `export JWT_TOKEN=<>`
- Assemble backsplash, start it up
- Make a request for a tile that has multiple underlying scenes:
`http :8081/dfc52922-dfc3-4155-9828-d8dfe9fd0f11/16/18192/24312/?token=${JWT_TOKEN}`
- Open up [Jaeger](http://localhost:16686/search) and view the trace to see what it looks like

Closes #5183
